### PR TITLE
Annual deficit recorders

### PR DESCRIPTION
**Pull request by MohammedBasheer**
AnnualCurtailmentRatioRecorder and AnnualSuppliedRatioRecorder previously crashed when max_flow equals zero

### CODE REVIEW

**Comment by jetuk** [comment]
@MohammedBasheer thanks for spotting the divide by zero error. I'll apply a fix to the other PR and merge that one. We can then close this one.

File: pywr/GIT_VERSION.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+5ebdecf8d129e1767622a9741eecde9d254c2564
```
Comment (snorfalorpagus) [inline pywr/GIT_VERSION.txt]: This file should not be commited to the repository.

File: pywr/recorders/_recorders.pyx [hunk 2]
```diff
@@ -810,7 +813,10 @@ cdef class AnnualCurtailmentRatioRecorder(AbstractAnnualRecorder):
 
         for scenario_index in self.model.scenarios.combinations:
             j = scenario_index.global_id
-            self._data[i, j] = 1 - self._actual_flow[i, j] / self._max_flow[i, j]
+            if self._max_flow[i, j] <= 0.0000000000001:
```
Comment (snorfalorpagus) [inline pywr/recorders/_recorders.pyx]: Use scientific notation for this, i.e. `1e-13`.

Is this consistent with other threadsholds we've used. i.e. Why not -12 or -14?

File: pywr/recorders/_recorders.pyx [hunk 3]
```diff
@@ -810,7 +813,10 @@ cdef class AnnualCurtailmentRatioRecorder(AbstractAnnualRecorder):
...
+                1
```
Comment (snorfalorpagus) [inline pywr/recorders/_recorders.pyx]: This is a no-op.

File: pywr/recorders/_recorders.pyx [hunk 4]
```diff
@@ -778,7 +778,10 @@ cdef class AnnualSuppliedRatioRecorder(AbstractAnnualRecorder):
 
         for scenario_index in self.model.scenarios.combinations:
             j = scenario_index.global_id
-            self._data[i, j] = self._actual_flow[i, j] / self._max_flow[i, j]
+            if self._max_flow[i, j] <= 0.0000000000001:
+                1
```
Comment (snorfalorpagus) [inline pywr/recorders/_recorders.pyx]: As below.

File: docs/source/cookbook/aggregated_node.rst [hunk 5]
```diff
@@ -0,0 +1,89 @@
+Aggregated nodes
```
Comment (snorfalorpagus) [inline docs/source/cookbook/aggregated_node.rst]: This documentation doesn't seem to be related to the code change. Could it be split into another MR?
